9.2-jre7/docker-entrypoint.sh

```diff
@@ -59,7 +59,9 @@ if expr "$*" : 'java .*/start\.jar.*$' >/dev/null ; then
 			--version |\
 			-v )\
 			# It is a terminating command, so exec directly
-			exec "$@"
+			JAVA="$1"
+			shift
+			exec $JAVA $JAVA_OPTIONS "$@"
```

Comment:
I feel like this script is getting pretty complicated and it would ideally be less so. One thought I have about this particular hunk is that it could make sense to extract a function that does this interpolation of `$JAVA_OPTIONS`. It also feels like that should happen right after the `# this is a command to run jetty` line, perhaps putting everything back into `$@` so that most of the script can still just `exec "$@"`?

Reply:
The separate function may not be necessary if the interpolation happens after that `# this is a command to run jetty` line, but I didn't attempt to make the changes myself

Reply:
I kind of agree, but will merge like this while I ponder a better way to simplify